### PR TITLE
hack/go-lint: update golanci-lint to v1.53.1

### PR DESCRIPTION
**Pull request by vincepri**
Related to https://github.com/openshift/release/pull/44928

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -20,14 +21,11 @@ linters:
   - asciicheck
   - containedctx
   - decorder
-  - depguard
```
Comment (vincepri) [inline .golangci.yaml:23]: This was never configured, and it defaults to stdlib only, which is not useful in this case

File: .golangci.yaml [hunk 2]
```diff
@@ -20,14 +21,11 @@ linters:
...
-  - dogsled
   - errcheck
   - errorlint
   - exportloopref
   - goconst
   - gocritic
-  - gocyclo
```
Comment (vincepri) [inline .golangci.yaml:30]: Not sure if this is very helpful at all, given that it counts the number of branches in a function/method to determine complexity, which might be a bit arbitrary

File: .golangci.yaml [hunk 3]
```diff
@@ -20,14 +21,11 @@ linters:
   - asciicheck
   - containedctx
   - decorder
-  - depguard
-  - dogsled
```
Comment (vincepri) [inline .golangci.yaml:24]: We have a lot of ignored parameters around

File: .golangci.yaml (hunk 2 above)
Comment (vincepri) [reply .golangci.yaml:30]: readded

**Comment by patrickdillon** [comment]
This LGTM. Will give rafael a chance to take a look.

File: .golangci.yaml (hunk 1 above)
Comment (r4f4) [reply .golangci.yaml:23]: :+1:  We can always bring it back when we need it.

File: .golangci.yaml [hunk 4]
```diff
@@ -20,20 +21,18 @@ linters:
...
   - goconst
   - gocritic
-  - gocyclo
   - godot
   - gofmt
   - goimports
   - gosec
   - gosimple
   - govet
+  - gocyclo
```
Comment (r4f4) [inline .golangci.yaml:35]: Nit: le'ts add it in alphabetical order.

File: .golangci.yaml (hunk 3 above)
Comment (r4f4) [reply .golangci.yaml:24]: That sounds like a reason to maybe keep it. Let me check what changed from the previous version.

Comment (vincepri) [reply .golangci.yaml:24]: Errors are checked elsewhere, but for some things it seems the returns are ignored on purpose

Comment (vincepri) [reply .golangci.yaml:24]: (errcheck checks return errors that are ignored)

File: .golangci.yaml (hunk 4 above)
Comment (vincepri) [reply .golangci.yaml]: done

File: .golangci.yaml (hunk 3 above)
Comment (r4f4) [reply .golangci.yaml:24]: If you are running the linter locally, are you passing `--new-from-rev=75173a17cf`?

File: .golangci.yaml [hunk 5]
```diff
@@ -11,6 +11,7 @@ run:
   - ^upi
   go: '1.19'
```
Comment (r4f4) [inline .golangci.yaml:12]: Maybe we can bump this to 1.20? We couldn't before because of the linter version.

Comment (vincepri) [reply .golangci.yaml]: done!

File: .golangci.yaml (hunk 3 above)
Comment (vincepri) [reply .golangci.yaml:24]: Readed `dogsled` and added the same commit we updated in https://github.com/openshift/release/pull/44928/files

**Pull request merged**